plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/TrinoNessieCatalog.java

```diff
@@ -0,0 +1,286 @@
+
+import static io.trino.plugin.iceberg.IcebergUtil.getIcebergTableWithMetadata;
+import static io.trino.plugin.iceberg.IcebergUtil.quotedTableName;
+import static io.trino.plugin.iceberg.IcebergUtil.validateTableCanBeDropped;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static java.util.Objects.requireNonNull;
+
+public class TrinoNessieCatalog
+        extends AbstractTrinoCatalog
+{
+    private final String warehouseLocation;
+    private final NessieIcebergClient nessieClient;
+    private final Map<SchemaTableName, TableMetadata> tableMetadataCache = new ConcurrentHashMap<>();
+
+    public TrinoNessieCatalog(
+            CatalogName catalogName,
+            TypeManager typeManager,
+            IcebergTableOperationsProvider tableOperationsProvider,
+            NessieIcebergClient nessieClient,
+            String warehouseLocation,
+            String trinoVersion,
+            boolean useUniqueTableLocation)
+    {
+        super(catalogName, typeManager, tableOperationsProvider, trinoVersion, useUniqueTableLocation);
+        this.warehouseLocation = requireNonNull(warehouseLocation, "warehouseLocation is null");
+        this.nessieClient = requireNonNull(nessieClient, "nessieClient is null");
+    }
+
+    @Override
+    public List<String> listNamespaces(ConnectorSession session)
+    {
+        return nessieClient.listNamespaces();
+    }
+
+    @Override
+    public void dropNamespace(ConnectorSession session, String namespace)
+    {
+        nessieClient.dropNamespace(namespace);
+    }
+
+    @Override
+    public Map<String, Object> loadNamespaceMetadata(ConnectorSession session, String namespace)
+    {
+        return nessieClient.loadNamespaceMetadata(namespace);
+    }
+
+    @Override
+    public Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session, String namespace)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void createNamespace(ConnectorSession session, String namespace, Map<String, Object> properties, TrinoPrincipal owner)
+    {
+        nessieClient.createNamespace(namespace, properties);
+    }
+
+    @Override
+    public void setNamespacePrincipal(ConnectorSession session, String namespace, TrinoPrincipal principal)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "setNamespacePrincipal is not supported for Iceberg Nessie catalogs");
+    }
+
+    @Override
+    public void renameNamespace(ConnectorSession session, String source, String target)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "renameNamespace is not supported for Iceberg Nessie catalogs");
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> namespace)
+    {
+        return nessieClient.listTables(namespace);
+    }
+
+    @Override
+    public Table loadTable(ConnectorSession session, SchemaTableName table)
+    {
+        TableMetadata metadata = tableMetadataCache.computeIfAbsent(
```

Comment:
Is there is any specific reason on why we didn't implement an eviction process here ?

Reply:
basically I aligned the implementation to how it is in `TrinoGlueCatalog`